### ISSUE TITLE
Texture needs to have the COPY_DST flag. Otherwise we can't ensure initialized memory!

### ISSUE THREAD

**Issue by simast** [gfx-rs/wgpu#2309]
**Description**
Texture creation fails with wgpu 0.12.

**Repro steps**

I have the following code which creates a frame texture that I then render the output to (for post processing):

```rust
let texture = device.create_texture(&TextureDescriptor {
    label: Some("Post-Processing Frame Texture"),
    size: Extent3d {
        width: surface_config.width,
        height: surface_config.height,
        depth_or_array_layers: 1,
    },
    mip_level_count: 1,
    sample_count: 1,
    dimension: TextureDimension::D2,
    format: surface_config.format,
    usage: TextureUsages::TEXTURE_BINDING
        | TextureUsages::COPY_DST // Added as part of wgpu 0.12 migration
        | TextureUsages::RENDER_ATTACHMENT,
});

```

However, since wgpu 0.12 update this is always hitting a debug assertion on the following line: https://github.com/gfx-rs/wgpu/blob/54f20bedef2db9f1c7806cb0878feb8f48206758/wgpu-core/src/command/memory_init.rs#L332-L333

**Expected vs observed behavior**
Texture creation works as expected.

**Platform**
Windows 11, Vulcan backend, wgpu 0.12.

**Comment by kvark** [comment]
Thank you for filing!
This is being addressed in #2307

**Comment by niklaskorz** [comment]
Fixed by #2307 :) can confirm it works now on https://github.com/niklaskorz/linon/tree/wgpu-0.12
